apps/studio/components/interfaces/DataWarehouse/CreateWarehouseCollection.tsx

```diff
@@ -0,0 +1,98 @@
+import { useParams } from 'common'
+import { useCreateCollection } from 'data/analytics/warehouse-collections-create-mutation'
+import { useRouter } from 'next/router'
+import React from 'react'
+import toast from 'react-hot-toast'
+import { Button, IconPlus, Input, Modal } from 'ui'
+
+type Props = {}
+
+export const CreateWarehouseCollectionModal = (props: Props) => {
+  const [isOpen, setIsOpen] = React.useState(false)
+  const router = useRouter()
+  const { ref } = useParams()
+
+  const {
+    mutateAsync: createCollection,
+    isLoading,
+    data: newCollection,
+  } = useCreateCollection({
+    projectRef: ref || 'default',
+    onSuccess: (data) => {
+      router.push(`/project/${ref}/logs/collections/${data.data?.token}`)
+    },
+  })
+
+  return (
+    <>
+      <Button
+        type="default"
+        className="justify-start flex-grow w-full"
+        icon={<IconPlus size="tiny" />}
+        onClick={() => {
+          setIsOpen(!isOpen)
+        }}
+      >
+        New collection
+      </Button>
+      <Modal
+        size="medium"
+        onCancel={() => setIsOpen(!isOpen)}
+        header="Create an event collection"
+        visible={isOpen}
+        hideFooter
+      >
+        <form
```

Comment:
can we use react hook form for consistency? ideally we just have one way of dealing with forms in the dashboard so its easier to maintain